### PR TITLE
[UI] As a logged in user, I can quit survey at any time

### PR DESCRIPTION
**Pull request by tyrro**
## What happened 👀
Closes https://github.com/tyrro/react-survey/issues/38

In this PR, a modal is implemented where users can opt-out of a survey at any time!

## Insight 📝
Modal is implemented to facilitate a user exit. at any time of the survey.

## Proof Of Work 📹

image

### CODE REVIEW

File: components/Button/index.jsx [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
 
-const Button = ({ size, dark = false, children, ...attributes }) => {
+const Button = ({ size, dark = false, className, children, ...attributes }) => {
   const sizeClass = SIZE_CLASSES[size];
   const backgroundClass = dark ? 'bg-dark text-white' : 'bg-white';
 
   return (
-    <button className={`text-base-large font-extrabold rounded-[10px] ${sizeClass} ${backgroundClass}`} {...attributes}>
+    <button
+      className={`text-base-large font-extrabold rounded-[10px] w-full ${className} ${sizeClass} ${backgroundClass}`}
```
[the remark this answers is not in the document]
Comment (tyrro) [reply components/Button/index.jsx:14]: I think there's an issue with the `font-weight` property on figma.
Although it's written as `font-weight: 400`, the button appears to be a `font-weight: 800`
I followed other projects for example: https://github.com/carryall/react-survey-challenge/blob/76e6c071f5b458644bd71d289209e6fa51f4f58e/src/assets/stylesheets/base/_buttons.scss#L2, and they are also following the `font-weight: 800` property for buttons, maybe I am missing here?

File: components/survey/Quit/index.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        data-test-id={surveyQuitTestIds.quitSurveyIcon}
+        onClick={() => setQuitConfirmationModalOpen(value => !value)}
+      />
+      {quitConfirmationModalOpen &&
+        createPortal(
+          <div
+            className="w-[313px] md:w-[390px] lg:w-[467px] absolute left-2/4 top-2/4 -translate-x-2/4 -translate-y-2/4 bg-sidebar/100 backdrop-blur-4xl rounded-[14px] p-6"
+            data-test-id={surveyQuitTestIds.base}
+          >
+            <div className="font-extrabold text-white text-base-xxl mb-2" data-test-id={surveyQuitTestIds.warningText}>
```
[the remark this answers is not in the document]
Comment (tyrro) [reply components/survey/Quit/index.jsx:36]: The text `Warning` has a `font-weight: 400` as you showed:

image

The next line with the text `Are you sure you want to quit the survey?` also has a `font-weight: 400` property, which is visibly lighter than the above text:

image

File: components/survey/Quit/index.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const [quitConfirmationModalOpen, setQuitConfirmationModalOpen] = useState(false);
+
+  return (
+    <>
+      <Image
+        className="cursor-pointer"
+        src={circleXmarkIcon}
+        alt="quit survey"
+        data-test-id={surveyQuitTestIds.quitSurveyIcon}
+        onClick={() => setQuitConfirmationModalOpen(value => !value)}
```
[the remark this answers is not in the document]
Comment (tyrro) [reply components/survey/Quit/index.jsx:37]: Liked the idea of treating the `x` icon for opening the modal only!

Therefore, I used the `useRootClose` hook from `react-overlay` in 6c46fb6aad0de63d8d6fef7ed71223b9207ca5ad, which handles

- closing modal when clicking outside it
- `ESC` button closes the modal too

File: components/survey/Quit/index.test.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      expect(warningText).toBeVisible();
+      expect(warningText).toHaveTextContent('Warning!');
+    });
+
+    it('renders a quit confirmation text', () => {
+      const quitConfirmationText = screen.getByTestId(surveyQuitTestIds.quitConfirmationText);
+
+      expect(quitConfirmationText).toBeVisible();
+      expect(quitConfirmationText).toHaveTextContent('Are you sure you want to quit the survey?');
+    });
```
[the remark this answers is not in the document]
Comment (tyrro) [reply components/survey/Quit/index.test.jsx:53]: sure,  done in bc310c7433775a55a7ab9b3d81abdf7a0ae1ff38!

**Review by hoangmirs — APPROVED** [review]

File: components/Button/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (carryall) [reply components/Button/index.jsx:14]: Ahh I think I found this issue before on some font files, if we installed the font on our machine, the `font-weight` we see on Figma would be a different value.
I've tried installing these fonts, and the bold `font-weight` turns to 850 😂 
- [Neuzeit S LT Std Book](https://www.download-free-fonts.com/details/73358/neuzeit-s-lt-std-book)
- [Neuzeit S LT Std Book Heavy](https://www.download-free-fonts.com/details/79166/neuzeit-s-lt-std-book-heavy)

Screen Shot 2565-05-10 at 18 04 47

but I'm not sure that if we use it, it'll look exactly the same as we expect or not 😅

**Pull request merged**